test/CareTogether.Core.Test/AuthorizationEngineTests/AuthorizeNoteCommandAsync.cs

```diff
@@ -0,0 +1,288 @@
+
+        private AuthorizationEngine? dut;
+        private Mock<IUserAccessCalculation>? mockUserAccessCalculation; // Change to interface
+
+        [TestInitialize]
+        public async Task TestInitializeAsync()
+        {
+            var accountsEventLog = new MemoryEventLog<AccountEvent>();
+            var personAccessEventLog = new MemoryEventLog<PersonAccessEvent>();
+            var directoryEventLog = new MemoryEventLog<DirectoryEvent>();
+            var goalsEventLog = new MemoryEventLog<GoalCommandExecutedEvent>();
+            var referralsEventLog = new MemoryEventLog<ReferralEvent>();
+            var approvalsEventLog = new MemoryEventLog<ApprovalEvent>();
+            var notesEventLog = new MemoryEventLog<NotesEvent>();
+            var communitiesEventLog = new MemoryEventLog<CommunityCommandExecutedEvent>();
+            var draftNotesStore = new MemoryObjectStore<string?>();
+            var configurationStore = new MemoryObjectStore<OrganizationConfiguration>();
+            var policiesStore = new MemoryObjectStore<EffectiveLocationPolicy>();
+            var organizationSecretsStore = new MemoryObjectStore<OrganizationSecrets>();
+
+            await TestDataProvider.PopulateTestDataAsync(
+                accountsEventLog,
+                personAccessEventLog,
+                directoryEventLog,
+                goalsEventLog,
+                referralsEventLog,
+                approvalsEventLog,
+                notesEventLog,
+                communitiesEventLog,
+                draftNotesStore,
+                configurationStore,
+                policiesStore,
+                organizationSecretsStore,
+                testSourceSmsPhoneNumber: null
+            );
+
+            var directoryResource = new DirectoryResource(directoryEventLog, Mock.Of<IFileStore>());
+            var policiesResource = new PoliciesResource(
+                configurationStore,
+                policiesStore,
+                organizationSecretsStore,
+                personAccessEventLog
+            );
+            var accountsResource = new AccountsResource(accountsEventLog, personAccessEventLog);
+
+            var notesResource = new NotesResource(notesEventLog, draftNotesStore);
+
+            // Replace UserAccess with IUserAccess mock:
+            mockUserAccessCalculation = new Mock<IUserAccessCalculation>();
+
+            dut = new AuthorizationEngine(
+                policiesResource,
+                directoryResource,
+                accountsResource,
+                notesResource,
+                mockUserAccessCalculation.Object // Use the interface mock here
+            );
+        }
+
+        [TestMethod]
+        public async Task UserHasNoAccessToNotes()
+        {
+            var user = PersonUserWithRoles(
+                Id('4'),
+                Guid.Parse("e3aaef77-0e97-47a6-b788-a67c237c781e")
+            );
+
+            mockUserAccessCalculation!
```

Comment:
Can you create a helper method for this pattern? Something like:
```c#
private MockUserAccess(params IEnumerable<Permission> permissions)
{
    mockUserAccessCalculation!
        .Setup(x => ...)
        .ReturnsAsync(ImmutableList.Create(permissions));
}
```

Reply:
Done